### PR TITLE
avoid Get data warning for auto-chart main object

### PR DESCRIPTION
**Pull request by DnlLrssn**
**Checklist**

- [ ] tests added
- [x] commits conform to the [Commit message format](./CONTRIBUTING.md#commit)
- [ ] documentation updated

**Squash commit message**

Avoid warnings for getting data on auto-chart main object.

### CODE REVIEW

File: session/autocharthandler.go [hunk 1]
```diff
@@ -23,9 +24,33 @@ func (handler *AutoChartHandler) Instance(id string) ObjectHandlerInstance {
...
+	instance.GetObjectLayout(sessionState, actionState, obj)
+	handleAutoChart(sessionState, actionState, genObj, obj)
+}
+
+// GetObjectLayout for auto-chart main object
+func (instance *AutoChartInstance) GetObjectLayout(sessionState *State, actionState *action.State, obj *enigmahandlers.Object) {
 	sessionState.QueueRequest(func(ctx context.Context) error {
-		return GetObjectLayout(sessionState, actionState, obj)
-	}, actionState, true, "")
+		enigmaObject, ok := obj.EnigmaObject.(*enigma.GenericObject)
```
Comment (SeBBBe) [inline session/autocharthandler.go:34]: Assume this is a different object than `genObj *enigma.GenericObject` in the call to `SetObjectAndEvents`?

Comment (DnlLrssn) [reply session/autocharthandler.go:34]: auto-chart main object will always be *enigma.GenericObject

**Comment by DnlLrssn** [comment]
> Don't really understand what this change does

auto-chart main object should only sen a GetLayout. This was previously achieved by having a object definition with auto-chart set to getLayout. without this definition and using the generic GetObjectLayout we would get a warning such as  `Get Data for object type<auto-chart> not supported`, protocolwise the correct messages where sent though. But since main auto-chart object only ever send GetLayout, and the rest of the logic is on the session object, this is a stripped-dpwn version of GetObjectLayout

File: session/autocharthandler.go (hunk 1 above)
Comment (DnlLrssn) [reply session/autocharthandler.go:34]: ah, understand the question now, no should be the same, logic was copied from the generic version, I could probably send in genObj instead, yes

**Comment by SeBBBe** [comment]
That makes sense. Thx for the explanation

File: session/autocharthandler.go (hunk 1 above)
Comment (SeBBBe) [reply session/autocharthandler.go:34]: Doesn't really matter. But that means this code and the error check are not strictly needed.

Comment (DnlLrssn) [reply session/autocharthandler.go:34]: you're right, didn't check it through for redundant things.

**Pull request merged**